### PR TITLE
SSPROD-26997: Changes to support key alias and global deployment

### PR DESCRIPTION
**Pull request by mdkulkarni15**

### CODE REVIEW

File: modules/services/agentless-scanning/variables.tf [hunk 1]
```diff
@@ -26,3 +26,13 @@ variable "tags" {
     "product" = "sysdig-secure-for-cloud"
   }
 }
+variable "deploy_global_resources" {
+  description = "(Optional) Set this field to 'true' to deploy EventBridge to an AWS Organization (Or specific OUs)"
```
Comment (ravinadhruve10) [inline modules/services/agentless-scanning/variables.tf:30]: Is this description as expected for agentless scanning feature?

**Review by ravinadhruve10 — COMMENTED** [review]
As per my understanding, rest of the change lgtm.

File: modules/services/agentless-scanning/variables.tf (hunk 1 above)
Comment (haresh-suresh) [reply modules/services/agentless-scanning/variables.tf:30]: Yeah this is definitely not right. We should just substitute, `Eventbridge` with `Agentless Scanning`.
cc: @nkraemer-sysdig

**Review by noahkraemer — CHANGES_REQUESTED** [review]
There are a few missing pieces here:

1) The `deploy_global_resources` flag needs to be set on other global resources (iam policies etc)
2) We need to create a [kms_replica_key](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/kms_replica_key) in each region
3) Variable description needs fixing as Ravina pointed out

File: modules/services/agentless-scanning/variables.tf (hunk 1 above)
Comment (ravinadhruve10) [reply modules/services/agentless-scanning/variables.tf]: Done, fixed the description.

**Comment by ravinadhruve10** [comment]
Added the relevant changes for resolving the missing pieces mentioned above.

File: modules/services/agentless-scanning/main.tf [hunk 2]
```diff
@@ -231,15 +243,25 @@ data "aws_iam_policy_document" "key_policy" {
...
   tags                    = var.tags
 }
 
-resource "aws_kms_alias" "agentless" {
-  name          = "alias/${var.name}"
-  target_key_id = aws_kms_key.agentless.key_id
+resource "aws_kms_replica_key" "agentless_replica" {
+  count = var.deploy_global_resources ? 0 : 1
+
+  description             = "Multi-Region replica key"
```
Comment (nkraemer-sysdig) [inline modules/services/agentless-scanning/main.tf:259]: Maybe add a `Sysdig` to this description?

**Review by noahkraemer — DISMISSED** [review]
Looking good!

File: modules/services/agentless-scanning/main.tf (hunk 2 above)
Comment (ravinadhruve10) [reply modules/services/agentless-scanning/main.tf]: yes, makes sense. Updated and fixed in next commit change.

**Review by noahkraemer — APPROVED** [review]

**Pull request merged**